### DISCUSSION TITLE
virustotal reports trojan

### DISCUSSION THREAD

**Discussion by harald-dunkel** [rancher-sandbox/rancher-desktop#6414] [General]
Current msi file version 1.12.2 linked from https://rancherdesktop.io/ contains the file Icon.RancherDesktopIcon.exe with the hash A896BCD5DB02C9BD68B4BCBF81187E7571314110F2D8C3B6A362452B9068592E
virustotal has 11 vendors saying this ist a trojan / malicious / unsafe.
Can anyone else confirm this?

**Comment by mook-as** [comment]
Relevant VirusTotal link: [`a896bcd5`](https://www.virustotal.com/gui/file/a896bcd5db02c9bd68b4bcbf81187e7571314110f2d8c3b6a362452b9068592e/detection) — that is indeed `Icon.RancherDesktopIcon.exe` referenced by [`Rancher.Desktop.Setup.1.12.2.msi`](https://www.virustotal.com/gui/file/7b55e4be321d463a799a9b8e4413409660557937415f007e659516ec54ed7511/details).

Of course, that file does [literally nothing](https://github.com/rancher-sandbox/rancher-desktop/blob/main/src/go/dummy/main.go); it's just there to provide an icon for the installer (because we previously used the main application executable for this, which was 150MB, and pointlessly bloated our installer).  Perhaps they don't like the fact that it does too little? For the next time, I can try to make it do something (or add an icon to [`rdctl.exe`](https://www.virustotal.com/gui/file/f615796403c2beaf5d6fb17c8e95ab55861e5312c5868c37a8ad7a9dae30d46d/details)) instead?

**Comment by harald-dunkel** [comment]
Thank you for your answer.
The linked file with the source code looks so tiny.
The Icon.RancherDesktopIcon.exe from the .msi file has 853 kByte

**Reply by mook-as** [reply]
Yep, it's basically all bloat from being compiled with golang, plus putting the icon in.
Feel free to check by building it yourself (yay open source), or grab a [build from CI](https://github.com/rancher-sandbox/rancher-desktop/actions/workflows/package.yaml?query=branch%3Av1.12.2) (on machines we don't own) and check the output.  We may end up finding a way to trim that down in the future, but that's not guaranteed (compared to the rest of the installer, this isn't that huge).

**Reply by jandubois** [reply]
It is possible to bring it down to about 5 kByte plus size of the icon, but probably not worth the effort (unless somebody wants to provide a PR): https://totallygamerjet.hashnode.dev/the-smallest-go-binary-5kb#

**Reply by harald-dunkel** [reply]
My problem is not really the size. My problem is the virus scanners that mark the file in version 1.12.1 and 1.12.2. So I have blocked two developer machines on which Rancher Desktop is installed and the virus scanner has reported problems. One machine can be restored from a backup. The other will probably have to be reinstalled at great expense. 

Because I have now seen that the file can become even larger if I compile it myself or even smaller. But I cannot say that there is no virus in the version that is in the .msi file they installed.

**Reply by harald-dunkel** [reply]
this is the file from 1.12.3: 
https://www.virustotal.com/gui/file/3cebc3608782a0b054a363b1fdb72aa0caf5058a2ac78f5ba6165fe69eadb836
it's getting worse

**Reply by harald-dunkel** [reply]
Is there any way of proving that the 12 AV vendors are wrong? Because with the CVEs that are fixed in 1.12.3, I can't let the developers continue working with the old versions. But I can't allow them to update to the new version either, because it potentially contains malicious code.

**Reply by jandubois** [reply]
You will have to take this up with the various AV vendors. The results seem bogus to me. I just cross-compiled an empty Go program on macOS for Windows and uploaded it to VT, and it shows 8 hits:

```console
$ cat main.go
package main;
func main() { }
$ GOOS=windows go build main.go
```

Then upload `main.exe` to VT and see this: https://www.virustotal.com/gui/file/98e8aed0863706e61b4ad9b455a7bf6eef46c755cbe8245b4448dfa3c9929525/detection

I am surprised that they claim both "Google" and "Microsoft" detect something there, but the only way this can have a virus/trojan would be if the Go runtime as released by Google itself has a virus, which seems unlikely.

Note that the `Icon.RancherDesktopIcon.exe` file was compiled on a Windows Github runner, and I compiled the empty file on a macOS system using cross-compilation. There is no shared software involved, except that both machines of course have the Go compiler and runtimes installed.

Occam's razor says the AV program results are bogus.

**Reply by jandubois** [reply]
Just throwing a couple links in here from a quick Google search:

https://go.dev/doc/faq#virus
https://www.reddit.com/r/golang/comments/10exyob/empty_func_main_is_detected_varius_virus_windows/
https://betterprogramming.pub/a-big-problem-in-go-that-no-one-talks-about-328cc3e71378

There are plenty more; maybe they will provide you with some peace of mind?

**Reply by harald-dunkel** [reply]
AV scanners do not recognize Occam's razor.
But they move files to quarantine.

**Comment by mook-as** [comment]
https://github.com/rancher-sandbox/rancher-desktop/pull/6430 filed to pick a (much larger!) executable, in the hopes that they will be satisfied by it actually doing something.

**Reply by harald-dunkel** [reply]
Thank you very much!
